cmd/root.go

```diff
@@ -51,6 +60,78 @@ func init() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 }
 
+// isArgTuneable returns if the argument provided is allowed to be modified
+func isArgTunable(arg string) bool {
+	return tuneableArgsWhitelist[arg]
+}
+
+// parseTuningFile parses the kernel argument tuning file
+func parseTuningFile(tuningFilePath string) (addArguments []types.TuneArgument, err error) {
+	if tuningFilePath == "" {
+		tuningFilePath = kernelTuningFile
+	}
+	// Return fast if the file does not exist
+	if _, err := os.Stat(tuningFilePath); os.IsNotExist(err) {
+		glog.V(2).Infof("no kernel tuning needed as %s does not exist", tuningFilePath)
+		// This isn't an error. Return out.
+		return addArguments, err
+	}
+	// Read and parse the file
+	file, err := os.Open(tuningFilePath)
+	// Clean up
+	defer func() {
+		closeErr := file.Close()
+		if closeErr != nil {
+			err = closeErr
+		}
+	}()
+	if err != nil {
+		// If we have an issue reading return an error
+		glog.Infof("Unable to open %s for reading: %v", tuningFilePath, err)
+		return addArguments, err
+	}
+
+	// Parse the tuning lines
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "ADD") {
+			// NOTE: Today only specific bare kernel arguments are allowed so
+			// there is not a need to split on =.
+			key := line[4:]
+			if isArgTunable(key) {
+				addArguments = append(addArguments, types.TuneArgument{Key: key, Bare: true})
+			} else {
+				glog.Infof("%s not a whitelisted kernel argument", key)
+			}
+		} else {
+			glog.V(2).Infof(`skipping malformed line in %s: "%s\n"`, tuningFilePath, line)
+		}
+	}
+	return addArguments, nil
+}
+
+// updateTuningArgs executes additions and removals of kernel tuning arguments
+func updateTuningArgs(tuningFilePath string) (changed bool, err error) {
+	changed = false
+	additions, err := parseTuningFile(tuningFilePath)
+	if err != nil {
+		return changed, err
+	}
+
+	// Execute additions
+	for _, toAdd := range additions {
+		if toAdd.Bare {
+			changed = true
+			utils.Run("rpm-ostree", "kargs", fmt.Sprintf("--append=%s", toAdd.Key))
```

Comment:
We could batch these in a single invocation. A bit moot since we only support one arg, so feel free to ignore for now. :) (Though maybe a `// TODO:` to that effect if we do extend on this?)

Reply:
There really only one invocation right now, but if this expands, absolutely!